In [1]:
import pandas as pd

pd.options.display.max_columns = None
pd.options.display.max_rows = None

# Current announcement dataset
df = pd.read_sas('pd_event_winsor.sas7bdat')

# Current completion dataset
df_comp = pd.read_sas('pd_completion_winsor.sas7bdat')

# Old completion dataset for comparison purposes
df_old = pd.read_sas('pd_comb_comp_lsa3.sas7bdat')
print(df.shape)

(332, 203)


Checking unique tickers in datasets to see if there's any difference

In [2]:
# Convert tickers into lists of strings
new_ticlist = [i.decode("utf-8") for i in df.tic.unique()]
old_ticlist = [i.decode("utf-8") for i in df_old.tic.unique()]
new_ticlist.sort()
old_ticlist.sort()
print(len(new_ticlist))

149


The two should be identical - difference should return an empty set

In [3]:
set(new_ticlist).difference(old_ticlist)

set()

Now that we know the two are identical, we can compare the 149 list to the 182 list

Convert ticker list of 182 acquirers supplied by Natalya into a list of strings

In [4]:
wrds = ['ABCB', 'AMNB', 'BAC', 'BANF', 'BBCN', 'BBT',
'BCBP', 'BDGE', 'BFIN', 'BHB', 'BHLB', 'BKYF', 'BMO', 
'BMRC', 'BMTC', 'BNCL', 'BNCN', 'BNS', 'BRKL', 'BTC', 
'CART', 'CBIN', 'CBSH', 'CBU', 'CCNE', 'CFFI', 'CFNL', 
'CHCO', 'CHFC', 'CLFC', 'CMA', 'COF', 'COLB', 'CSBC', 
'CSFL', 'CTBI', 'CVBF', 'CVCY', 'CYN', 'DNBK', 'EBSB',
'EFSC', 'ESSA', 'EVBN', 'EWBC', 'FBMS', 'FBNC', 'FCAL', 
'FCBC', 'FCCO', 'FCCY', 'FCLF', 'FCNCA', 'FFBC', 'FFCH', 
'FFIN', 'FFKY', 'FHN', 'FITB', 'FMBI', 'FMER', 'FNB', 
'FNFG', 'FPFC', 'FRME', 'FSFG', 'FUBC', 'GABC', 'GBCI', 
'GSBC', 'HBAN', 'HBCP', 'HBHC', 'HBNC', 'HBOS', 'HEOP', 
'HFWA', 'HIG', 'HMST', 'HNBC', 'HOMB', 'HOME', 'HTBI', 
'HTH', 'HTLF', 'IBKC', 'IBTX', 'INDB', 'ISBC', 'JFBI', 
'JPM', 'KFFB', 'KRNY', 'LARK', 'LBAI', 'LBCP', 'LCNB', 
'LION', 'MBFI', 'MBTF', 'MFSF', 'MSFG', 'MSL', 'MTB', 
'NBBC', 'NBTB', 'NBTF', 'NEBS', 'NHTB', 'NRIM', 'NWFL', 
'NYB', 'OKSB', 'OLBK', 'ONB', 'OSHC', 'OZRK', 'PACW', 
'PB', 'PBCT', 'PBSK', 'PCBK', 'PEBO', 'PFBI', 'PNC', 
'PPBI', 'PRSP', 'PSTB', 'PVTB', 'PWOD', 'QCRH', 'RBCAA', 
'RF', 'RIVR', 'RNST', 'ROMA', 'SASR', 'SAVB', 'SF', 'SFNC', 
'SIFI', 'SMBC', 'SNV', 'SONA', 'STBA', 'STBZ', 'STI', 'STSA',
'SUSQ', 'SYBT', 'TCBK', 'TD', 'THFF', 'THRD', 'TMP', 'TOBC',
'TOWN', 'TRMK', 'UBCP', 'UBNK', 'UBSH', 'UBSI', 'UMBF',
'UMPQ', 'USB', 'VBFC', 'VIST', 'VLY', 'VPFG', 'WABC',
'WAFD', 'WAL', 'WBCO', 'WFC', 'WFSL', 'WIBC', 'WSBC',
'WSFS', 'WTFC', 'WTNY', 'XBKS', 'ZION']


How many of the supplied tickers are missing from ticlist?

In [5]:
missings = set(wrds).difference(new_ticlist)
len(missings)

33

In [6]:
oldmissings = set(wrds).difference(old_ticlist)
len(missings)

33

Which tickers are missing?

In [7]:
missings

{'BANF',
 'BBCN',
 'BFIN',
 'BMO',
 'BNS',
 'BTC',
 'CBIN',
 'CBSH',
 'CTBI',
 'FCCO',
 'FCLF',
 'FFKY',
 'FITB',
 'FPFC',
 'GBCI',
 'HBNC',
 'HIG',
 'LBCP',
 'NEBS',
 'NHTB',
 'NYB',
 'OSHC',
 'PBSK',
 'PRSP',
 'QCRH',
 'RBCAA',
 'RF',
 'SNV',
 'STBZ',
 'STI',
 'TD',
 'VPFG',
 'WFSL'}

## Delisting codes for 7 of the 33 (182 - 149)

This portion deals with the 33 tickers missing between the original sample and the Kamakura merge.

Data also available in wrds_output_182.xlsx. 


| PERMNO | date      | TICKER | Company Name                     | Date of Delisting Payment | Delisting Code |
|--------|-----------|--------|----------------------------------|---------------------------|----------------|
| 11513  | 31-Dec-14 | FFKY   | FIRST FINANCIAL SERVICE CORP     | 2-Jan-15                  | 231            |
| 10638  | 30-Apr-10 | HNBC   | HARLEYSVILLE NATIONAL CORP PA    | 12-Apr-10                 | 231            |
| 91025  | 30-Nov-12 | NEBS   | NEW ENGLAND BANCSHARES INC       | 19-Nov-12                 | 231            |
| 91400  | 30-Sep-16 | FCLF   | FIRST CLOVER LEAF FINANCIAL CORP | 9-Sep-16                  | 241            |
| 90487  | 30-Nov-16 | OSHC   | OCEAN SHORE HOLDING CO NEW       | 1-Dec-16                  | 241            |
| 91404  | 31-Mar-10 | LBCP   | LIBERTY BANCORP INC MO           | 3-Mar-10                  | 570            |
| 86574  | 30-Nov-11 | FPFC   | FIRST PLACE FINANCIAL CORP NM    | 10-Nov-11                 | 584            |

The remaining 26 don't have a delisting code.

**Delisting Code/Meaning**

231	When merged, shareholders primarily receive common stock or ADRs. Replaces codes 201, 202 and 203. Codes 201-203 are no longer assigned.

241	When merged, shareholders primarily receive common stock and cash, issue on CRSP file.

570	Delisted by current exchange - company request (no reason given).

584	Delisted by current exchange - does not meet exchange’s financial guidelines for continued listing.

## Delisting codes for 32 of the 149

This portion deals with the remaining 149 tickers. Of the 149, these 32 have delisting codes and dates between 1/2007 and 12/2016.

Data also available in wrds_output_149.xlsx.

| PERMNO | date      | TICKER | Company Name                     | Date of Delisting Payment | Delisting Code |
|--------|-----------|--------|----------------------------------|---------------------------|----------------|
| 89524  | 30-Nov-11 | CLFC   | CENTER FINANCIAL CORP            | 1-Dec-11                  | 231            |
| 85980  | 28-Sep-12 | CSBC   | CITIZENS SOUTH BANKING CORP DEL  | 1-Oct-12                  | 231            |
| 91897  | 31-May-13 | FCAL   | FIRST CALIFORNIA FINL GROUP INC  | 3-Jun-13                  | 231            |
| 35503  | 31-Jul-13 | FFCH   | FIRST FINANCIAL HOLDINGS INC     | 29-Jul-13                 | 231            |
| 93040  | 31-Oct-14 | FUBC   | 1ST UNITED BANCORP INC           | 3-Nov-14                  | 231            |
| 90730  | 30-Jun-15 | HBOS   | HERITAGE FINANCIAL GROUP INC     | 1-Jul-15                  | 231            |
| 10638  | 30-Apr-10 | HNBC   | HARLEYSVILLE NATIONAL CORP PA    | 12-Apr-10                 | 231            |
| 89440  | 29-Feb-16 | NBBC   | NEWBRIDGE BANCORP                | 1-Mar-16                  | 231            |
| 91407  | 31-Dec-13 | ROMA   | ROMA FINANCIAL CORP              | 9-Dec-13                  | 231            |
| 76435  | 31-Dec-12 | SAVB   | SAVANNAH BANCORP INC             | 13-Dec-12                 | 231            |
| 80768  | 31-Oct-14 | THRD   | T F FINANCIAL CORP               | 27-Oct-14                 | 231            |
| 90824  | 30-Apr-14 | UBNK   | UNITED FINANCIAL BANCORP INC MD  | 1-May-14                  | 231            |
| 79584  | 31-Aug-12 | VIST   | V I S T FINACIAL CORP            | 2-Aug-12                  | 231            |
| 86440  | 29-Jul-16 | WIBC   | WILSHIRE BANCORP INC             | 1-Aug-16                  | 231            |
| 90295  | 29-Jul-16 | XBKS   | XENITH BANKSHARES INC            | 1-Aug-16                  | 231            |
| 76653  | 31-Oct-08 | AHG    | APRIA HEALTHCARE GROUP INC       | 29-Oct-08                 | 233            |
| 20694  | 28-Nov-08 | UB     | UNIONBANCAL CORP                 | 4-Nov-08                  | 233            |
| 92851  | 30-Jun-15 | BKYF   | BANK OF KENTUCKY FINANCIAL CORP  | 22-Jun-15                 | 241            |
| 23916  | 30-Oct-15 | CYN    | CITY NATIONAL CORP               | 2-Nov-15                  | 241            |
| 92522  | 30-Jun-11 | DNBK   | DANVERS BANCORP INC              | 1-Jul-11                  | 241            |
| 35167  | 31-Aug-16 | FMER   | FIRSTMERIT CORP                  | 16-Aug-16                 | 241            |
| 85994  | 29-Jul-16 | FNFG   | FIRST NIAGARA FINL GROUP INC NEW | 1-Aug-16                  | 241            |
| 90482  | 30-May-14 | HOME   | HOME FEDERAL BANCORP INC MD      | 19-May-14                 | 241            |
| 89798  | 30-May-14 | JFBI   | JEFFERSON BANCSHARES INC TENN    | 2-Jun-14                  | 241            |
| 89758  | 31-Mar-15 | NBTF   | N B & T FINANCIAL GROUP INC      | 9-Mar-15                  | 241            |
| 42307  | 30-Apr-08 | NX     | QUANEX CORP                      | 24-Apr-08                 | 241            |
| 84331  | 29-Feb-16 | RIVR   | RIVER VALLEY BANCORP             | 1-Mar-16                  | 241            |
| 11056  | 30-Apr-14 | STSA   | STERLING FINANCIAL CORP WASH     | 21-Apr-14                 | 241            |
| 73809  | 31-Jul-15 | SUSQ   | SUSQUEHANNA BANCSHARES INC PA    | 3-Aug-15                  | 241            |
| 92920  | 29-Feb-12 | TOBC   | TOWER BANCORP INC                | 21-Feb-12                 | 241            |
| 86199  | 30-Apr-14 | WBCO   | WASHINGTON BANKING COMPANY       | 1-May-14                  | 241            |
| 77053  | 30-Jun-11 | WTNY   | WHITNEY HOLDING CORP             | 7-Jun-11                  | 241            |

**Delisting Code/Meaning**

231	When merged, shareholders primarily receive common stock or ADRs. Replaces codes 201, 202 and 203. Codes 201-203 are no longer assigned.

233	When merged, shareholders receive cash payments.

241	When merged, shareholders primarily receive common stock and cash, issue on CRSP file.